### PR TITLE
Refine SignV2 enforcement and CU benchmarks

### PR DESCRIPTION
**Pull request by edo-chan**
## Summary

This PR now covers the full SignV2 improvement backlog from the Claude audit follow-up, rebased on current `main`.

Key changes:
- keep the existing SignV2 enforcement cleanup: unrestricted `All` / `AllButManageAuthority` fast path, flattened restricted-role enforcement, explicit snapshot cap, TransferChecked destination-limit coverage, and CU comparison harness
- clean up `process_instruction` / `execute` account handling with explicit first-account failure, initialized-prefix comments for the `MaybeUninit` buffers, one remaining-account capacity check, and flatter account classification
- simplify Swig config, stake, token, and program-scope classification while preserving hot-path direct indexing
- add parser bounds for `CreateV1`, `AddAuthorityV1`, `UpdateAuthorityV1`, and `SubAccountSignV1` length-delimited payloads
- validate duplicate/conflicting role actions on create/add/update paths, including fixed-vs-recurring limits that would otherwise compete for the same runtime enforcement slot
- add parser-bound unit tests, duplicate-action state tests, and add/update authority integration coverage

## Current LiteSVM CU Comparison

Delta is `new - baseline`; negative means cheaper. These are from `cargo test -p swig --test sign_v2_cu_comparison -- --nocapture` after the expanded backlog changes.

| Case | Baseline | New | Delta |
| --- | ---: | ---: | ---: |
| All | 3211 | 2856 | -355 CU (-11.06%) |
| SOL limit only | 4042 | 4125 | +83 CU (+2.05%) |
| SOL limit + destination | 4645 | 4760 | +115 CU (+2.48%) |
| Token limit only | 9489 | 9559 | +70 CU (+0.74%) |
| Token limit + destination | 10196 | 10386 | +190 CU (+1.86%) |

Interpretation:
- unrestricted `All` remains cheaper because it exits before restricted-role snapshots and limit accounting
- generic-only SOL/token paths are still close to baseline
- the added cost is concentrated in stricter destination/account parsing and account classification safety checks

## Verification

- `cargo fmt`
- `git diff --check`
- `cargo test -p swig-state validate_unique_role_actions -- --nocapture`
- `cargo test -p swig from_instruction_bytes_rejects -- --nocapture`
- `cargo test -p swig --test add_authority_test --test update_authority_test`
- `cargo test -p swig --test sign_v2 --test sol_destination_limit_v2 --test sol_recurring_destination_limit_v2 --test token_destination_limit_v2 --test token_recurring_destination_limit_v2`
- `cargo test -p swig --test sign_v2_cu_comparison -- --nocapture`
- `cargo build-sbf --arch v1`

### CODE REVIEW

File: state/src/swig.rs [hunk 1]
```diff
@@ -98,6 +98,115 @@ pub fn sub_account_signer<'a>(
...
+
+#[inline(always)]
+fn role_action_key<'a>(
+    permission: Permission,
+    action_data: &'a [u8],
+) -> Result<RoleActionKey<'a>, ProgramError> {
+    const PUBKEY_LEN: usize = 32;
+    const PROGRAM_SCOPE_TARGET_OFFSET: usize = 80;
+
+    let (kind, start, len) = match permission {
```
Comment (tracy-codes) [inline state/src/swig.rs]: `role_action_key` maps fixed and recurring variants for the same scope to the same key, so a role can no longer have both `TokenLimit` and `TokenRecurringLimit` for the same mint, both `SolLimit` and `SolRecurringLimit`, or both destination fixed/recurring limits. That is a behavior change at creation/update time, not just a SignV2 enforcement optimization.

If the intent is only to prevent ambiguous duplicate fixed limits or duplicate recurring limits, the key should include the concrete permission. If the intent is to make fixed and recurring mutually exclusive, this needs to be explicitly documented as a breaking semantics change and covered in client-facing tests.

**Review by tracy-codes — CHANGES_REQUESTED** [review]

File: state/src/swig.rs (hunk 1 above)
Comment (edo-chan) [reply state/src/swig.rs]: I think I am gonna do a similar pass to add authority in the future. And not make any hasty changes for now. Reverting!

**Review by tracy-codes — APPROVED** [review]
